From c45eb08e02acfa4ee86d6528cae9e0e726c0f23f Mon Sep 17 00:00:00 2001
From: Bhushan Shah <bhush94@gmail.com>
Date: Sat, 16 Dec 2017 11:00:02 +0000
Subject: [PATCH] Add includes copy in include subdir

I am not entirely sure why include dir is shipped in tarball
---
 .../private/qgstreamerelementcontrol_p.h           | 63 ++++++++++++++++++++
 .../private/qgstreamervideosinkcontrol_p.h         | 69 ++++++++++++++++++++++
 2 files changed, 132 insertions(+)
 create mode 100644 include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamerelementcontrol_p.h
 create mode 100644 include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamervideosinkcontrol_p.h

diff --git a/include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamerelementcontrol_p.h b/include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamerelementcontrol_p.h
new file mode 100644
index 0000000..69fd4a3
--- /dev/null
+++ b/include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamerelementcontrol_p.h
@@ -0,0 +1,63 @@
+/****************************************************************************
+**
+** Copyright (C) 2013 Jolla Ltd.
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia.  For licensing terms and
+** conditions see http://qt.digia.com/licensing.  For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights.  These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 3.0 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU General Public License version 3.0 requirements will be
+** met: http://www.gnu.org/copyleft/gpl.html.
+**
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#ifndef QGSTREAMERELEMENTCONTROL_P_H
+#define QGSTREAMERELEMENTCONTROL_P_H
+
+#include <QtMultimedia/qmediacontrol.h>
+
+typedef struct _GstElement GstElement;
+
+class QGStreamerElementControl : public QMediaControl
+{
+    Q_OBJECT
+public:
+    virtual ~QGStreamerElementControl();
+
+    virtual void setElement(GstElement *element) = 0;
+
+protected:
+    QGStreamerElementControl(QObject *parent = 0);
+};
+
+#define QGStreamerVideoSinkControl_iid "org.qt-project.qt.gstreamervideosinkcontrol/5.2"
+
+#endif
diff --git a/include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamervideosinkcontrol_p.h b/include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamervideosinkcontrol_p.h
new file mode 100644
index 0000000..c23d76f
--- /dev/null
+++ b/include/QtMultimedia/5.9.3/QtMultimedia/private/qgstreamervideosinkcontrol_p.h
@@ -0,0 +1,69 @@
+/****************************************************************************
+**
+** Copyright (C) 2013 Jolla Ltd.
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia.  For licensing terms and
+** conditions see http://qt.digia.com/licensing.  For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights.  These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 3.0 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU General Public License version 3.0 requirements will be
+** met: http://www.gnu.org/copyleft/gpl.html.
+**
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#ifndef QGSTREAMERVIDEOSINKCONTROL_P_H
+#define QGSTREAMERVIDEOSINKCONTROL_P_H
+
+#include <private/qgstreamerelementcontrol_p.h>
+#include <private/qgstreamervideorendererinterface_p.h>
+
+class QGStreamerVideoSinkControl
+        : public QGStreamerElementControl
+        , public QGstreamerVideoRendererInterface
+{
+    Q_OBJECT
+    Q_INTERFACES(QGstreamerVideoRendererInterface)
+public:
+    QGStreamerVideoSinkControl(QObject *parent = 0);
+    ~QGStreamerVideoSinkControl();
+
+    GstElement *videoSink();
+    void setElement(GstElement *element);
+
+Q_SIGNALS:
+    void sinkChanged();
+    void readyChanged(bool);
+
+private:
+    GstElement *m_videoSink;
+};
+
+#endif
-- 
2.7.4

